ignite/contrib/handlers/hpo_logger.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+import collections
+from collections import defaultdict, UserString
+from collections.abc import Mapping, Sequence
+from enum import Enum
+import json
+import os
+from threading import RLock
+import time
+from typing import Any
+
+from ignite.engine import Engine, Events
+
+_DEFAULT_METRIC_PATH = '/tmp/hypertune/output.metrics'
+
+_MAX_NUM_METRIC_ENTRIES_TO_PRESERVE = 100
+
+def _get_loss_from_output(output: Sequence[Mapping[str, Any]]) -> Any:
+    return output[0]["loss"]
+
+class MetricLoggerKeys(Enum):
+    METRICS = "Metrics"
+    LOSS = "Loss"
+
+
+class HPOLogger(object):
+    """
+    Makes selected metric accessible for use by GCP Vertex AI hyperparameter tuning jobs. By adding only this
+    single HPO-logger it unlocks hyperparam screening for Ignite
+
+    Example::
+        # construct an evaluator saving metric values
+        val_metrics = {"accuracy": Accuracy(), "nll": Loss(criterion)}
+        evaluator = create_supervised_evaluator(model, metrics=val_metrics, device=device)
+        evaluator.logger = setup_logger("evaluator")
+
+        # construct the logger and associate with evaluator to extract metric values from
+        hpo_logger = HPOLogger(evaluator = evaluator, metric_tag= 'nll')
+
+        # construct the trainer with the logger passed in as a handler so that it logs loss values
+        trainer.add_event_handler(Events.EPOCH_COMPLETED, hpo_logger)
+
+        # run training
+        trainer.run(train_loader, max_epochs=epochs)
+
+    Args:
+        evaluator: Evaluator to consume metric results from at the end of its evaluation run
+        metric_tag: Converts the metric value coming from the trainer/evaluator's state into a storable value
+    """
+
+    def __init__(
+        self,
+        evaluator: Engine | None = None,
+        metric_tag: UserString = 'training/hptuning/metric'
```

Comment:
Can't we use `str` here ?
```suggestion
        metric_tag: str = 'training/hptuning/metric'
```